core/domain/collection_domain.py

```diff
@@ -401,19 +472,23 @@ def serialize(self):
         # files must add a version parameter to their files with the correct
         # version of this object. The line below must then be moved to
         # to_dict().
-        collection_dict['version'] = self.version
+        # Since, collectionDict is a Dict representation of latest schema
+        # domain object and it does not contain version, created_on and
+        # last_updated keys. Thus to prevent MyPy errors, we added an ignore
+        # here.
+        collection_dict['version'] = self.version # type: ignore[misc]
 
         if self.created_on:
-            collection_dict['created_on'] = (
+            collection_dict['created_on'] = ( # type: ignore[misc]
```

Comment:
Why is this ignore needed?

Reply:
As we defined TypedDict of collection domain object to match with current version of it. we are considering fields which are defined in `to_dict`, `from_dict` and latest schema version. So, collection_dict's type is collectionDict and collectionDict does not contain `created_on`, `last_updated` and `version` field. which cause MyPy to throw error. So, just to prevent it from error i added this ignore stmt. 

comment is also added above for the same.

Reply:
Hmm, should we rather introduce new type for this and then retype it?

Reply:
Here what i think :
1. to_dict() method returns different typeDict ( which does not contain created_on, last_updated and version field) and here collection_dict expecting a typedict which contain all fields. So, to assignment of these different types require an ignore stmt.
2. also while defining the type for this we should inherit from collectionDict. So that there is no duplicate code exists.
3. while defining the fields we have to  make  created_on and last_updated to be of type str.

I created the new type and assigned it!, by keeping the above points in mind.